make cortex-arm fail on master (57a5315) and v1.0.5

cannot make that instructions work, particularly the `make arm-cortex` step https://dev.zenroom.org/#/pages/cortex

I am in a debian 10 stable 4.19.0-13-amd64

installed dependencies

```
apt-get install -y git build-essential cmake zsh xxd gcc-arm-none-eabi
```

build instructions:

```
git clone https://github.com/dyne/Zenroom.git
cd Zenroom
#git tag 
git checkout v1.0.5
git submodule update --init --recursive
make cortex-arm
```

error details:

```
[ 30%] Linking C static library lib/libamcl_bls_BLS461.a
/usr/bin/cmake -P CMakeFiles/amcl_bls_BLS461.dir/cmake_clean_target.cmake
/usr/bin/cmake -E cmake_link_script CMakeFiles/amcl_bls_BLS461.dir/link.txt --verbose=1
/usr/bin/ar qc lib/libamcl_bls_BLS461.a  CMakeFiles/amcl_bls_BLS461.dir/src/bls_BLS461.c.obj
/usr/bin/arm-none-eabi-ranlib lib/libamcl_bls_BLS461.a
make[3]: Leaving directory '/home/user/dev/Zenroom/lib/milagro-crypto-c'
[ 30%] Built target amcl_bls_BLS461
make -f test/CMakeFiles/test_bls_sss_BLS461.dir/build.make test/CMakeFiles/test_bls_sss_BLS461.dir/depend
make[3]: Entering directory '/home/user/dev/Zenroom/lib/milagro-crypto-c'
cd /home/user/dev/Zenroom/lib/milagro-crypto-c && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /home/user/dev/Zenroom/lib/milagro-crypto-c /home/user/dev/Zenroom/lib/milagro-crypto-c/test /home/user/dev/Zenroom/lib/milagro-crypto-c /home/user/dev/Zenroom/lib/milagro-crypto-c/test /home/user/dev/Zenroom/lib/milagro-crypto-c/test/CMakeFiles/test_bls_sss_BLS461.dir/DependInfo.cmake --color=
Dependee "/home/user/dev/Zenroom/lib/milagro-crypto-c/test/CMakeFiles/test_bls_sss_BLS461.dir/DependInfo.cmake" is newer than depender "/home/user/dev/Zenroom/lib/milagro-crypto-c/test/CMakeFiles/test_bls_sss_BLS461.dir/depend.internal".
Dependee "/home/user/dev/Zenroom/lib/milagro-crypto-c/test/CMakeFiles/CMakeDirectoryInformation.cmake" is newer than depender "/home/user/dev/Zenroom/lib/milagro-crypto-c/test/CMakeFiles/test_bls_sss_BLS461.dir/depend.internal".
Scanning dependencies of target test_bls_sss_BLS461
make[3]: Leaving directory '/home/user/dev/Zenroom/lib/milagro-crypto-c'
make -f test/CMakeFiles/test_bls_sss_BLS461.dir/build.make test/CMakeFiles/test_bls_sss_BLS461.dir/build
make[3]: Entering directory '/home/user/dev/Zenroom/lib/milagro-crypto-c'
[ 31%] Building C object test/CMakeFiles/test_bls_sss_BLS461.dir/test_bls_sss_BLS461.c.obj
cd /home/user/dev/Zenroom/lib/milagro-crypto-c/test && /usr/bin/arm-none-eabi-gcc -DCMAKE -I/home/user/dev/Zenroom/lib/milagro-crypto-c/include  -DARCH_CORTEX -mcpu=cortex-m4 -mthumb -mlittle-endian -mthumb-interwork -Wstack-usage=1024 -DLIBRARY -Wno-main -ffreestanding -nostartfiles -std=c99 -static-libgcc -Wall -Wextra -Wno-strict-prototypes -Wunused-value -Wcast-align -Wunused-variable -Wundef -Wformat-security -Wshadow -O2   -o CMakeFiles/test_bls_sss_BLS461.dir/test_bls_sss_BLS461.c.obj   -c /home/user/dev/Zenroom/lib/milagro-crypto-c/test/test_bls_sss_BLS461.c
/home/user/dev/Zenroom/lib/milagro-crypto-c/test/test_bls_sss_BLS461.c: In function 'main':
/home/user/dev/Zenroom/lib/milagro-crypto-c/test/test_bls_sss_BLS461.c:242:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
/home/user/dev/Zenroom/lib/milagro-crypto-c/test/test_bls_sss_BLS461.c: In function 'test':
/home/user/dev/Zenroom/lib/milagro-crypto-c/test/test_bls_sss_BLS461.c:45:5: warning: stack usage might be 4512 bytes [-Wstack-usage=]
 int test(csprng *RNG)
     ^~~~
[ 32%] Linking C executable ../bin/test_bls_sss_BLS461
cd /home/user/dev/Zenroom/lib/milagro-crypto-c/test && /usr/bin/cmake -E cmake_link_script CMakeFiles/test_bls_sss_BLS461.dir/link.txt --verbose=1
/usr/bin/arm-none-eabi-gcc  -DARCH_CORTEX -mcpu=cortex-m4 -mthumb -mlittle-endian -mthumb-interwork -Wstack-usage=1024 -DLIBRARY -Wno-main -ffreestanding -nostartfiles -std=c99 -static-libgcc -Wall -Wextra -Wno-strict-prototypes -Wunused-value -Wcast-align -Wunused-variable -Wundef -Wformat-security -Wshadow -O2   CMakeFiles/test_bls_sss_BLS461.dir/test_bls_sss_BLS461.c.obj  -o ../bin/test_bls_sss_BLS461 ../lib/libamcl_bls_BLS461.a ../lib/libamcl_pairing_BLS461.a ../lib/libamcl_curve_BLS461.a ../lib/libamcl_core.a 
/usr/lib/gcc/arm-none-eabi/7.3.1/../../../arm-none-eabi/bin/ld: warning: cannot find entry symbol _start; defaulting to 0000000000008000
/usr/lib/gcc/arm-none-eabi/7.3.1/../../../arm-none-eabi/bin/ld: /usr/lib/gcc/arm-none-eabi/7.3.1/../../../arm-none-eabi/lib/thumb/v7e-m/libc.a(lib_a-sbrkr.o): in function `_sbrk_r':
/tmp/building/package/build/arm-none-eabi/thumb/v7e-m/newlib/libc/reent/../../../../../../../newlib/libc/reent/sbrkr.c:51: undefined reference to `_sbrk'
/usr/lib/gcc/arm-none-eabi/7.3.1/../../../arm-none-eabi/bin/ld: /usr/lib/gcc/arm-none-eabi/7.3.1/../../../arm-none-eabi/lib/thumb/v7e-m/libc.a(lib_a-writer.o): in function `_write_r':
/tmp/building/package/build/arm-none-eabi/thumb/v7e-m/newlib/libc/reent/../../../../../../../newlib/libc/reent/writer.c:49: undefined reference to `_write'
/usr/lib/gcc/arm-none-eabi/7.3.1/../../../arm-none-eabi/bin/ld: /usr/lib/gcc/arm-none-eabi/7.3.1/../../../arm-none-eabi/lib/thumb/v7e-m/libc.a(lib_a-closer.o): in function `_close_r':
/tmp/building/package/build/arm-none-eabi/thumb/v7e-m/newlib/libc/reent/../../../../../../../newlib/libc/reent/closer.c:47: undefined reference to `_close'
/usr/lib/gcc/arm-none-eabi/7.3.1/../../../arm-none-eabi/bin/ld: /usr/lib/gcc/arm-none-eabi/7.3.1/../../../arm-none-eabi/lib/thumb/v7e-m/libc.a(lib_a-fstatr.o): in function `_fstat_r':
/tmp/building/package/build/arm-none-eabi/thumb/v7e-m/newlib/libc/reent/../../../../../../../newlib/libc/reent/fstatr.c:55: undefined reference to `_fstat'
/usr/lib/gcc/arm-none-eabi/7.3.1/../../../arm-none-eabi/bin/ld: /usr/lib/gcc/arm-none-eabi/7.3.1/../../../arm-none-eabi/lib/thumb/v7e-m/libc.a(lib_a-isattyr.o): in function `_isatty_r':
/tmp/building/package/build/arm-none-eabi/thumb/v7e-m/newlib/libc/reent/../../../../../../../newlib/libc/reent/isattyr.c:52: undefined reference to `_isatty'
/usr/lib/gcc/arm-none-eabi/7.3.1/../../../arm-none-eabi/bin/ld: /usr/lib/gcc/arm-none-eabi/7.3.1/../../../arm-none-eabi/lib/thumb/v7e-m/libc.a(lib_a-lseekr.o): in function `_lseek_r':
/tmp/building/package/build/arm-none-eabi/thumb/v7e-m/newlib/libc/reent/../../../../../../../newlib/libc/reent/lseekr.c:49: undefined reference to `_lseek'
/usr/lib/gcc/arm-none-eabi/7.3.1/../../../arm-none-eabi/bin/ld: /usr/lib/gcc/arm-none-eabi/7.3.1/../../../arm-none-eabi/lib/thumb/v7e-m/libc.a(lib_a-readr.o): in function `_read_r':
/tmp/building/package/build/arm-none-eabi/thumb/v7e-m/newlib/libc/reent/../../../../../../../newlib/libc/reent/readr.c:49: undefined reference to `_read'
collect2: error: ld returned 1 exit status
make[3]: *** [test/CMakeFiles/test_bls_sss_BLS461.dir/build.make:88: bin/test_bls_sss_BLS461] Error 1
make[3]: Leaving directory '/home/user/dev/Zenroom/lib/milagro-crypto-c'
make[2]: *** [CMakeFiles/Makefile2:1284: test/CMakeFiles/test_bls_sss_BLS461.dir/all] Error 2
make[2]: Leaving directory '/home/user/dev/Zenroom/lib/milagro-crypto-c'
make[1]: *** [Makefile:163: all] Error 2
make[1]: Leaving directory '/home/user/dev/Zenroom/lib/milagro-crypto-c'
make: *** [Makefile:115: milagro] Error 2
```

--- a/Makefile
+++ b/Makefile
@@ -115,13 +115,15 @@ milagro:
 	@echo "-- Building milagro (${system})"
 	if ! [ -r ${pwd}/lib/milagro-crypto-c/CMakeCache.txt ]; then \
 		cd ${pwd}/lib/milagro-crypto-c && \
+		mkdir -p build && \
+		cd build && \
 		CC=${gcc} LD=${ld} \
-		cmake . -DCMAKE_C_FLAGS="${cflags}" -DCMAKE_SYSTEM_NAME="${system}" \
+		cmake ../ -DCMAKE_C_FLAGS="${cflags}" -DCMAKE_SYSTEM_NAME="${system}" \
 		-DCMAKE_AR=${ar} -DCMAKE_C_COMPILER=${gcc} ${milagro_cmake_flags}; \
 	fi
 	if ! [ -r ${pwd}/lib/milagro-crypto-c/lib/libamcl_core.a ]; then \
 		CC=${gcc} CFLAGS="${cflags}" AR=${ar} RANLIB=${ranlib} LD=${ld} \
-		make -C ${pwd}/lib/milagro-crypto-c; \
+		make -C ${pwd}/lib/milagro-crypto-c/build; \
 	fi
 
 check-milagro: milagro
@@ -157,11 +159,12 @@ install-lua:
 
 clean:
 	make clean -C ${pwd}/lib/lua53/src
-	if [ -f ${pwd}/lib/milagro-crypto-c/Makefile ]; then \
-		make clean -C ${pwd}/lib/milagro-crypto-c; \
+	if [ -f ${pwd}/lib/milagro-crypto-c/build/Makefile ]; then \
+		make clean -C ${pwd}/lib/milagro-crypto-c/build; \
 	fi
 	rm -f ${pwd}/lib/milagro-crypto-c/CMakeCache.txt
 	rm -rf ${pwd}/lib/milagro-crypto-c/CMakeFiles
+	rm -rf ${pwd}/lib/milagro-crypto-c/build
 	git checkout ${pwd}/lib/milagro-crypto-c/CPackConfig.cmake
 	make clean -C ${pwd}/src
 	make clean -C ${pwd}/bindings

--- a/build/config.mk
+++ b/build/config.mk
@@ -9,7 +9,7 @@ website := ${pwd}/docs
 luasrc := ${pwd}/lib/lua53/src
 ldadd := ${pwd}/lib/lua53/src/liblua.a
 lua_embed_opts := ""
-lua_cflags := -DLUA_COMPAT_5_3 -DLUA_COMPAT_MODULE -DLUA_COMPAT_BITLIB -I${pwd}/lib/milagro-crypto-c/include -I${pwd}/src
+lua_cflags := -DLUA_COMPAT_5_3 -DLUA_COMPAT_MODULE -DLUA_COMPAT_BITLIB -I${pwd}/lib/milagro-crypto-c/build/include -I${pwd}/src -I${pwd}/lib/milagro-crypto-c/include 
 
 # ----------------
 # zenroom defaults
@@ -33,7 +33,7 @@ rsa_bits := ""
 ecdh_curve := SECP256K1
 ecp_curve  := BLS383
 milagro_cmake_flags := -DBUILD_SHARED_LIBS=OFF -DBUILD_PYTHON=OFF -DBUILD_DOXYGEN=OFF -DBUILD_DOCS=OFF -DBUILD_BENCHMARKS=OFF -DBUILD_EXAMPLES=OFF -DWORD_SIZE=32 -DBUILD_PAILLIER=OFF -DBUILD_X509=OFF -DBUILD_WCC=OFF -DBUILD_MPIN=OFF -DAMCL_CURVE=${ecdh_curve},${ecp_curve} -DAMCL_RSA=${rsa_bits} -DCMAKE_SHARED_LIBRARY_LINK_FLAGS="" -DC99=1 -DPAIRING_FRIENDLY_BLS383='BLS' -DCOMBA=1
-milib := ${pwd}/lib/milagro-crypto-c/lib
+milib := ${pwd}/lib/milagro-crypto-c/build/lib
 ldadd += ${milib}/libamcl_curve_${ecp_curve}.a
 ldadd += ${milib}/libamcl_pairing_${ecp_curve}.a
 ldadd += ${milib}/libamcl_curve_${ecdh_curve}.a

--- a/src/Makefile
+++ b/src/Makefile
@@ -22,7 +22,7 @@ ARCH := $(shell uname -m)
 BRANCH := $(shell git symbolic-ref HEAD | sed -e 's,.*/\(.*\),\1,')
 HASH := $(shell git rev-parse --short HEAD)
 VERSION := ${ZENROOM_VERSION}+${HASH}
-CFLAGS  += -I. -I../lib/lua53/src -I../lib/milagro-crypto-c/include -Wall -Wextra
+CFLAGS  += -I. -I../lib/lua53/src -I../lib/milagro-crypto-c/build/include -I../lib/milagro-crypto-c/include -Wall -Wextra
 SOURCES := \
 	cli.o jutils.o zenroom.o zen_error.o \
 	lua_functions.o lua_modules.o lualibs_detected.o lua_shims.o \

